src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Validation/ValidationVisitor.cs

```diff
@@ -283,6 +289,23 @@ protected virtual bool Visit(ModelMetadata metadata, string key, object model)
                 CurrentPath.Pop(model);
                 return true;
             }
+            // If the metadata indicates that no validators exist AND the aggregate state for the key says that the model graph
+            // is not invalid (i.e. is one of Unvalidated, Valid, or Skipped) we can safely mark the graph as valid.
+            else if (
+                AllowShortCircuitingValidationWhenNoValidatorsArePresent &&
+                metadata.HasValidators == false &&
+                ModelState.GetFieldValidationState(key) != ModelValidationState.Invalid)
+            {
+                // No validators will be created for this graph of objects. Mark it as valid if it wasn't previously validated.
+                var entries = ModelState.FindKeysWithPrefix(key);
+                foreach (var item in entries)
+                {
+                    if (item.Value.ValidationState == ModelValidationState.Unvalidated)
+                    {
+                        item.Value.ValidationState = ModelValidationState.Valid;
+                    }
+                }
+            }
```

Comment:
This code block duplicates line 276-271 - I'd expect to see a single line diff here (L295)